src/providers/auth-service/auth-service.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@angular/core';
+import { Http } from '@angular/http';
+import 'rxjs/add/operator/map';
+import {Md5} from "ts-md5/dist/md5";
+
+//@Models
+import {User} from "../../models/user";
+import {Credentials} from "../../models/credentials";
+
+/*
+  Generated class for the AuthServiceProvider provider.
+
+  See https://angular.io/docs/ts/latest/guide/dependency-injection.html
+  for more info on providers and Angular DI.
+*/
+@Injectable()
+export class AuthServiceProvider {
+
+  constructor() {
+  }
+
+  public authUser(c : Credentials){
+    let r = null;
```

Comment:
null is not the same that undefined. Here, you need to declarate a undefined value and you can do something like this: `let r;`